opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -279,11 +285,19 @@ impl LogProcessor for BatchLogProcessor {
         }
 
         let result = self
-            .message_sender
-            .try_send(BatchMessage::ExportLog(Box::new((
-                record.clone(),
-                instrumentation.clone(),
-            ))));
+            .logs_sender
+            .try_send(Box::new((record.clone(), instrumentation.clone())));
+
+        if self.current_batch_size.fetch_add(1, Ordering::Relaxed) >= self.max_export_batch_size {
```

Comment:
This will add small perf overhead with user thread. I believe this should be find to have, given the advantage this allow us to wake up the processor-thread only when export is required. But good to have some measure of the cost.

Reply:
I ran the stress test and benchmarks, and the performance seems to have improved. I believe it's because the logger threads incur lesser synchronization costs as their receiver isn't actively working on the channel. I updated the PR description with the setup and results. Let me know what you think. It would be great if you could verify it on your machine as well.

Reply:
Yes, I see similar improvement with stress test with your branch - 17.2 iterations/sec (main) to 20.5 iterations/sec.